packages/flutter/lib/src/widgets/widget_span.dart

```diff
@@ -0,0 +1,254 @@
+///       height: 50,
+///       child: Card(
+///         child: Center(
+///           child: Text('Hello World!')
+///         )
+///       ),
+///     )
+///   ),
+///   TextSpan(text: 'the best!'),
+/// ]
+/// ```
+/// {@end-tool}
+///
+/// The [semanticsLabel] argument may be provided to supply a description of
+/// this [WidgetSpan] within the [toPlainText]. The semantics label of the
+/// overall [InlineSpan] tree will not mention [WidgetSpan]s unless
+/// [semanticsLabel] is provided. Semantics for the widget itself will still
+/// function independently from the [InlineSpan] semantics label.
+///
+/// See also:
+///
+///  * [TextSpan], a node that represents text in a [InlineSpan] tree.
+///  * [Text], a widget for showing uniformly-styled text.
+///  * [RichText], a widget for finer control of text rendering.
+///  * [TextPainter], a class for painting [InlineSpan] objects on a [Canvas].
+@immutable
+class WidgetSpan extends PlaceholderSpan {
+  /// Creates a [WidgetSpan] with the given values.
+  ///
+  /// The [child] property should be non-null. [WidgetSpan] is a leaf node in
+  /// the [InlineSpan] tree
+  ///
+  /// A [TextStyle] may be provided with the [style] property, but only the
+  /// decoration, foreground, background, and spacing options will be used.
+  const WidgetSpan({
+    @required this.child,
+    InlineWidgetAlignment alignment = InlineWidgetAlignment.bottom,
+    TextBaseline baseline,
+    TextStyle style,
+  }) : assert(child != null),
+       assert((alignment == InlineWidgetAlignment.aboveBaseline ||
+               alignment == InlineWidgetAlignment.belowBaseline ||
+               alignment == InlineWidgetAlignment.baseline) ? baseline != null : true),
+       super(alignment:
+           // Convert InlineWidgetAlignment to PlaceholderAlignment in a const fashion.
+           alignment == InlineWidgetAlignment.baseline ? ui.PlaceholderAlignment.baseline :
+           alignment == InlineWidgetAlignment.aboveBaseline ? ui.PlaceholderAlignment.aboveBaseline :
+           alignment == InlineWidgetAlignment.belowBaseline ? ui.PlaceholderAlignment.belowBaseline :
+           alignment == InlineWidgetAlignment.top ? ui.PlaceholderAlignment.top :
+           alignment == InlineWidgetAlignment.bottom ? ui.PlaceholderAlignment.bottom :
+           alignment == InlineWidgetAlignment.middle ? ui.PlaceholderAlignment.middle : null,
+         baseline: baseline, style: style,
+       );
+
+  /// The widget to embed inline within text.
+  final Widget child;
+
+  /// Adds a placeholder box to the paragraph builder if a size has been
+  /// calculated for the widget.
+  ///
+  /// Sizes are provided through [dimensions], which should contain a 1:1
+  /// in-order mapping of widget to laid out dimensions. If no such dimension
+  /// is provided, the widget will be skipped.
+  ///
+  /// Since widget sizes are calculated independently from the rest of the
+  /// paragraph, the [textScaleFactor] is ignored.
```

Comment:
you know, one could argue that we should honour the textScaleFactor by just transforming each child by the given factor...

Reply:
Since we are deferring all sizing, offsetting, positioning, etc to the child widget, I figured it would be more consistent to not apply scaling here and allow the users to scale it themselves in the child widget tree.

Reply:
My concern is that most people don't even know about textScaleFactor. It's an accessibility feature. This means that most people won't honor it, and the result will be that users that need it will find apps are unusable because while the text will grow, images within the text (or whatever) will not.

It should be relatively straight-forward to just apply a transform at paint time to zoom up the child widgets though, no? We just lay them out like normal, then tell the TextPainter that they need to be twice as big as they said they were going to be (or whatever textScaleFactor is), then when we paint we scale them up (also when we hit test we'd adjust the offset appropriately).